diff -up v8-3.2.10/src/runtime.cc.always-false v8-3.2.10/src/runtime.cc
--- v8-3.2.10/src/runtime.cc.always-false	2011-06-10 11:05:03.000000000 -0400
+++ v8-3.2.10/src/runtime.cc	2011-06-10 11:20:09.713526215 -0400
@@ -4092,8 +4092,8 @@ RUNTIME_FUNCTION(MaybeObject*, Runtime_S
   StrictModeFlag strict_mode = kNonStrictMode;
   if (args.length() == 5) {
     CONVERT_SMI_CHECKED(strict_unchecked, args[4]);
-    RUNTIME_ASSERT(strict_unchecked == kStrictMode ||
-                   strict_unchecked == kNonStrictMode);
+    // RUNTIME_ASSERT(strict_unchecked == kStrictMode ||
+    //               strict_unchecked == kNonStrictMode);
     strict_mode = static_cast<StrictModeFlag>(strict_unchecked);
   }
 
