diff -up v8-4.10.91/Makefile.system_icu v8-4.10.91/Makefile
--- v8-4.10.91/Makefile.system_icu	2016-01-27 17:20:20.554136527 -0500
+++ v8-4.10.91/Makefile	2016-01-27 17:21:03.377907390 -0500
@@ -44,6 +44,10 @@ endif
 ifdef component
   GYPFLAGS += -Dcomponent=$(component)
 endif
+# system_icu=on
+ifeq ($(system_icu), on)
+  GYPFLAGS += -Duse_system_icu=1
+endif
 # disassembler=on
 ifeq ($(disassembler), on)
   GYPFLAGS += -Dv8_enable_disassembler=1
