diff -up v8-5.2.258/Makefile.bb v8-5.2.258/Makefile
--- v8-5.2.258/Makefile.bb	2016-10-10 14:57:17.275968725 -0400
+++ v8-5.2.258/Makefile	2016-10-10 14:58:37.012406994 -0400
@@ -37,6 +37,12 @@ NACL_SDK_ROOT ?=
 
 # Special build flags. Use them like this: "make library=shared"
 
+# bundledbinutils=off
+ifeq ($(bundledbinutils), off)
+  GYPFLAGS += -Dlinux_use_bundled_binutils=0
+else
+  GYPFLAGS += -Dlinux_use_bundled_binutils=1
+endif
 # library=shared || component=shared_library
 ifeq ($(library), shared)
   GYPFLAGS += -Dcomponent=shared_library
