diff -up v8-6.2.91/gypfiles/standalone.gypi.ppc64 v8-6.2.91/gypfiles/standalone.gypi
--- v8-6.2.91/gypfiles/standalone.gypi.ppc64	2017-07-31 11:24:31.000000000 -0400
+++ v8-6.2.91/gypfiles/standalone.gypi	2017-07-31 11:50:42.467600076 -0400
@@ -268,7 +268,7 @@
       }, {
         'v8_enable_gdbjit%': 0,
       }],
-      ['(OS=="linux" or OS=="mac") and (target_arch=="ia32" or target_arch=="x64") and \
+      ['(OS=="linux" or OS=="mac") and (target_arch=="ia32" or target_arch=="x64" or target_arch=="ppc64") and \
         v8_target_arch!="x32"', {
         'clang%': 1,
       }, {
@@ -788,7 +788,7 @@
             or v8_target_arch=="mips64el")', {
             'cflags': [ '-Wshorten-64-to-32' ],
           }],
-          [ 'host_arch=="ppc64" and OS!="aix"', {
+          [ 'host_arch=="ppc64" and OS!="aix" and clang==0', {
             'cflags': [ '-mminimal-toc' ],
           }],
           [ 'visibility=="hidden" and v8_enable_backtrace==0', {
