diff -up v8-6.2.91/tools/gn/bootstrap/bootstrap.py.fabi11 v8-6.2.91/tools/gn/bootstrap/bootstrap.py
--- v8-6.2.91/tools/gn/bootstrap/bootstrap.py.fabi11	2018-02-19 17:11:22.650679128 -0500
+++ v8-6.2.91/tools/gn/bootstrap/bootstrap.py	2018-02-19 17:11:46.579128407 -0500
@@ -314,6 +314,11 @@ def write_gn_ninja(path, root_gen_dir, o
   cflags = os.environ.get('CFLAGS', '').split()
   cflags_cc = os.environ.get('CXXFLAGS', '').split()
   ldflags = os.environ.get('LDFLAGS', '').split()
+
+  # Work around GCC8 bug gcc#84286
+  cflags.extend(['-fabi-version=11'])
+  cflags_cc.extend(['-fabi-version=11'])
+
   include_dirs = [root_gen_dir, SRC_ROOT]
   libs = []
 
