diff -up v8-6.2.91/build/config/compiler/BUILD.gn.memexhaust v8-6.2.91/build/config/compiler/BUILD.gn
--- v8-6.2.91/build/config/compiler/BUILD.gn.memexhaust	2017-08-03 17:15:41.355834826 -0400
+++ v8-6.2.91/build/config/compiler/BUILD.gn	2017-08-03 17:15:46.587704667 -0400
@@ -1739,7 +1739,7 @@ config("symbols") {
         # TODO(thakis): Remove this again once dump_syms is fixed.
         cflags += [ "-gdwarf-3" ]
       }
-      cflags += [ "-g2" ]
+      cflags += [ "-g1" ]
     }
     if (use_debug_fission) {
       cflags += [ "-gsplit-dwarf" ]
