diff -up v8-6.2.91/build/config/compiler/BUILD.gn.nounused v8-6.2.91/build/config/compiler/BUILD.gn
--- v8-6.2.91/build/config/compiler/BUILD.gn.nounused	2017-08-04 10:35:44.212100721 -0400
+++ v8-6.2.91/build/config/compiler/BUILD.gn	2017-08-04 10:36:25.554097933 -0400
@@ -1740,6 +1740,7 @@ config("symbols") {
         cflags += [ "-gdwarf-3" ]
       }
       cflags += [ "-g2" ]
+      cflags += [ "-Wno-unused-function" , "-Wno-unused-result" ]
     }
     if (use_debug_fission) {
       cflags += [ "-gsplit-dwarf" ]
